python/ray/experimental/channel/shared_memory_channel.py

```diff
@@ -156,9 +160,9 @@ def __init__(
         reader_and_node_list: List[Tuple["ray.actor.ActorHandle", str]],
         typ: Optional[Union[int, SharedMemoryType]] = None,
         _writer_node_id: Optional["ray.NodeID"] = None,
-        _reader_node_id: Optional["ray.NodeID"] = None,
+        _reader_node_ids: Optional[Set["ray.NodeID"]] = None,
         _writer_ref: Optional["ray.ObjectRef"] = None,
-        _reader_ref: Optional["ray.ObjectRef"] = None,
+        _reader_refs: Optional[Dict[str, "ray.ObjectRef"]] = None,
```

Comment:
Can you please update docstring?

Reply:
Yeah I think we should move it to the docstring so that the caller knows what to pass in. There are also args that seem to contain overlapping information, which need clean up or clarification.